capstone/src/main/java/com/google/sps/servlets/ChallengeServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.Objects.Challenge;
+
+@WebServlet("challenge")
+public class ChallengeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Challenge").addSort("dueDate", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    // Get most recent challenge in database
+    Entity entity = results.asIterable().iterator().next();
+    Challenge challenge = Challenge.fromEntity(entity);
+    String json = new Gson().toJson(challenge);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
```

Comment:
Long-term since this is repeated in EVERY servlet, I'd love to see a way to manage this for all servlets without having to repeat these three lines.

Reply:
Made changes in new pull request (mult users)-- created ServletHelper class